web/src/message_fetch.js

```diff
@@ -428,9 +429,33 @@ export function maybe_load_older_messages(opts) {
     if (!msg_list_data.fetch_status.can_load_older_messages()) {
         // We may already be loading old messages or already
         // got the oldest one.
+        if (opts.recent_view) {
+            recent_view_ui.set_backfill_in_progress(false);
+        }
         return;
     }
 
+    if (opts.recent_view) {
+        let fetched_substantial_history = false;
+        if (msg_list_data.num_items() >= consts.recent_view_minimum_load_more_fetch_size) {
+            fetched_substantial_history = true;
+        }
```

Comment:
I'm not entirely sure that this algorithm is ideal in the event that we've already got a lot of history, but it might feel OK.